memoryallocation.h

```diff
@@ -25,10 +25,24 @@
 #include <cstdlib>
 #include <cstddef>
 #include <stdexcept>
+#include <string.h>
+
 #ifdef USE_JEMALLOC
 #include "jemalloc/jemalloc.h"
 #endif
 
+#ifndef NDEBUG
+#ifndef INITIALIZE_ALIGNED_MALLOC_WITH_NAN
+#define INITIALIZE_ALIGNED_MALLOC_WITH_NAN
```

Comment:
I think this init-with-NaN flag would be useful in dev as well? Could you please set up a PR for that as well?

Reply:
Already done here in dev:
https://github.com/fmihpc/vlasiator/pull/691/files